src/digests/digest.lisp

```diff
@@ -177,17 +177,18 @@
              (let ((digest-size ,(if single-digest-p
                                      (second (first specs))
                                      `(etypecase state
-                                        ,@(reverse specs)))))
+                                        ,@(reverse specs))))
+                   (state-copy (copy-digest state)))
```

Comment:
I think having ```produce-digest``` return the correct hash every time would be a convenient feature, and the cost would only be copying a few hundred bytes at most.

But anyway if you think it would hinder performances too much, we can revert commit b8317e02587544ec3b0b55a8eb5e3f2c39c12dee.